apps/testing-javascript/src/components/certificate-form.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react'
+import * as Dialog from '@radix-ui/react-dialog'
+import {track} from '@skillrecordings/skill-lesson/utils/analytics'
+import {DownloadIcon, XIcon} from '@heroicons/react/solid'
+import {useSession} from 'next-auth/react'
+import {useForm} from 'react-hook-form'
+
+const CertificateForm: React.FC<React.PropsWithChildren> = () => {
+  const {data: session} = useSession()
+  const form = useForm({
+    shouldUseNativeValidation: true,
+    defaultValues: {
+      name: session?.user?.name,
+      certificateUrl: `/api/certificate?name=${session?.user?.name}`,
+    },
+  })
+  const {register, watch} = form
+  const handleNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    return form.setValue(
+      'certificateUrl',
+      `/api/certificate?name=${e.currentTarget.value || session?.user?.name}`,
+    )
+  }
+  React.useEffect(() => {
+    if (session?.user?.name) {
+      form.setValue('name', session?.user?.name)
+      form.setValue(
+        'certificateUrl',
+        `/api/certificate?name=${session?.user?.name}`,
+      )
+    }
+  }, [session])
```

Comment:
`session.user.name` gets `undefined` on the first render so used `useEffect` to update form values